test/session.test.js

```diff
@@ -0,0 +1,160 @@
+/* eslint-disable no-unused-expressions */
+/* eslint-disable no-undef */
+import * as Factory from './lib/factory.js';
+chai.use(chaiAsPromised);
+const expect = chai.expect;
+
+describe('server session', () => {
+  const BASE_ITEM_COUNT = 1; /** Default items key */
+
+  const syncOptions = {
+    checkIntegrity: true,
+    awaitAll: true
+  };
+
+  before(async function () {
+    localStorage.clear();
+  });
+
+  after(async function () {
+    localStorage.clear();
+  });
+
+  beforeEach(async function () {
+    this.timeout(Factory.TestTimeout);
+    this.expectedItemCount = BASE_ITEM_COUNT;
+    this.application = await Factory.createInitAppWithRandNamespace();
+    this.email = Uuid.GenerateUuidSynchronously();
+    this.password = Uuid.GenerateUuidSynchronously();
+    await Factory.registerUserToApplication({
+      application: this.application,
+      email: this.email,
+      password: this.password
+    });
+  });
+
+  afterEach(async function () {
+    expect(this.application.syncService.isOutOfSync()).to.equal(false);
+    const items = this.application.itemManager.items;
+    expect(items.length).to.equal(this.expectedItemCount);
+    const rawPayloads = await this.application.storageService.getAllRawPayloads();
+    expect(rawPayloads.length).to.equal(this.expectedItemCount);
+    await this.application.deinit();
+  });
+
+  function getDelayBeforeNextRequest(currentSession) {
```

Comment:
I think it would be more clear if you created a resuable async function called `sleepUntilSessionExpires`. It's really not clear by its title what this function does. The new function would calculate session expirey time, and also do the sleeping. This way callers don't have to separate the two.